driver-core/src/main/java/com/datastax/driver/core/EventDebouncer.java

```diff
@@ -205,14 +211,21 @@ private void scheduleDelayedDelivery() {
   }
 
   private DeliveryAttempt cancelDelayedDelivery() {
-    DeliveryAttempt previous = delayedDelivery.get();
+    return cancelDelivery(delayedDelivery.get());
+  }
+
+  private DeliveryAttempt cancelImmediateDelivery() {
```

Comment:
I think it doesn't hurt to cancel the immediate delivery as well, but the real fix is to actually catch ISE inside `SchemaParser.refresh()`.